### PR TITLE
Reserve vector storage in the convert filter code

### PR DESCRIPTION
**Pull request by pozdnyakov**

### CODE REVIEW

File: src/mbgl/style/conversion/function.cpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ optional<PropertyExpression<T>> convertFunctionToExpression(const Convertible& v
         return nullopt;
     }
 
-    optional<T> defaultValue;
+    optional<T> defaultValue{};
```
Comment (kkaefer) [inline src/mbgl/style/conversion/function.cpp]: is this just a cosmetic change or is that actually changing the behavior?

Comment (pozdnyakov) [reply src/mbgl/style/conversion/function.cpp]: that's fixing a compiler warning:
```
cc1plus: error: ‘defaultValue’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../../../src/mbgl/style/conversion/function.cpp:81:17: note: ‘defaultValue’ was declared here
     optional<T> defaultValue;
```

Comment (kkaefer) [reply src/mbgl/style/conversion/function.cpp]: Ah, it's initializing `defaultValue` with the default constructor of `T`.

Comment (pozdnyakov) [reply src/mbgl/style/conversion/function.cpp]: as from https://en.cppreference.com/w/cpp/experimental/optional/optional
`1) Constructs the object that does not contain a value.`

Comment (pozdnyakov) [reply src/mbgl/style/conversion/function.cpp]: though we can be more explicit and use
`optional<T> defaultValue{nullopt}`

Comment (kkaefer) [reply src/mbgl/style/conversion/function.cpp]: How is that fixing the compiler warning then?

Comment (pozdnyakov) [reply src/mbgl/style/conversion/function.cpp]: not sure :) this looks like a false warning as putting `{}` just calls the optional's default constructor, i.e. does not change anything

**Pull request merged**